### PR TITLE
Add 'package' property to ExtensionMetadata and deprecate 'software'

### PR DESCRIPTION
**Pull request by eslavich**
This renames the extension metadata `software` property to `package` to match the ASDF standard.  Files written by older versions of the library will read successfully, but emit a deprecation warning.

Addresses https://github.com/spacetelescope/asdf/issues/709

### CODE REVIEW

File: asdf/asdf.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
 from .exceptions import AsdfDeprecationWarning
 from .extension import AsdfExtensionList, default_extensions
 
-from .tags.core import AsdfObject, Software, HistoryEntry, ExtensionMetadata
+from .tags.core import AsdfObject, Software, HistoryEntry, ExtensionMetadata, Software
```
Comment (perrygreenfield) [inline asdf/asdf.py]: OK, I give up. How does importing Software twice change things? :-)

File: asdf/tags/core/__init__.py [hunk 2]
```diff
@@ -25,9 +26,17 @@ class ExtensionMetadata(AsdfType):
     name = 'core/extension_metadata'
     version = '1.0.0'
 
-    def __init__(self, extension_class=None, software={}):
+    def __init__(self, extension_class=None, package=None, software=None):
+        if software is not None:
+            warnings.warn("Previous versions of the ASDF library stored a 'software' property " +
+                          "in extension metadata.  That property has been renamed to 'package' " +
```
Comment (perrygreenfield) [inline asdf/tags/core/__init__.py]: The use of the term property here appears ambiguous. It wasn't strictly a Python property previously, though it is now.

File: asdf/asdf.py (hunk 1 above)
Comment (eslavich) [reply asdf/asdf.py]: Ha!  It's, uh, a test for code review quality.  Like Van Halen's brown M&Ms.

File: asdf/tags/core/__init__.py (hunk 2 above)
Comment (eslavich) [reply asdf/tags/core/__init__.py]: Good point, would "attribute" or "field" be better?

**Comment by nden** [comment]
I am going to merge this so we can see the jwst regression tests tonight.

File: asdf/tags/core/__init__.py (hunk 2 above)
Comment (perrygreenfield) [reply asdf/tags/core/__init__.py]: Yes, probably. Maybe something like "...saved this information in the .software attribute. That value is now saved in the .package attribute"

**Pull request merged**